delta-lake/src/integration/java/org/apache/iceberg/delta/TestSnapshotDeltaLakeTable.java

```diff
@@ -0,0 +1,496 @@
+                spark, newTableIdentifier, partitionedLocation)
+            .tableLocation(newIcebergTableLocation)
+            .execute();
+
+    checkSnapshotIntegrity(partitionedLocation, partitionedIdentifier, newTableIdentifier, result);
+    checkIcebergTableLocation(newTableIdentifier, newIcebergTableLocation);
+  }
+
+  @Test
+  public void testSnapshotWithAdditionalProperties() {
+    // add some properties to the original delta table
+    spark.sql(
+        "ALTER TABLE "
+            + unpartitionedIdentifier
+            + " SET TBLPROPERTIES ('foo'='bar', 'test0'='test0')");
+    String newTableIdentifier = destName(icebergCatalogName, snapshotAdditionalPropertiesTableName);
+    SnapshotDeltaLakeTable.Result result =
+        DeltaLakeToIcebergMigrationSparkIntegration.snapshotDeltaLakeTable(
+                spark, newTableIdentifier, unpartitionedLocation)
+            .tableProperty("test1", "test1")
+            .tableProperties(
+                ImmutableMap.of(
+                    "test2", "test2", "test3", "test3", "test4",
+                    "test4")) // add additional iceberg table properties
+            .execute();
+
+    checkSnapshotIntegrity(
+        unpartitionedLocation, unpartitionedIdentifier, newTableIdentifier, result);
+    checkIcebergTableLocation(newTableIdentifier, unpartitionedLocation);
+    checkIcebergTableProperties(
+        newTableIdentifier,
+        ImmutableMap.of(
+            "foo", "bar", "test0", "test0", "test1", "test1", "test2", "test2", "test3", "test3",
+            "test4", "test4"),
+        unpartitionedLocation);
+  }
+
+  @Test
+  public void testSnapshotTableWithExternalDataFiles() {
+    // Add parquet files to default.external_data_files_table. The newly added parquet files
+    // are not at the same location as the table.
+    addExternalDatafiles(externalDataFilesTableLocation, unpartitionedLocation);
+
+    String newTableIdentifier = destName(icebergCatalogName, snapshotExternalDataFilesTableName);
+    SnapshotDeltaLakeTable.Result result =
+        DeltaLakeToIcebergMigrationSparkIntegration.snapshotDeltaLakeTable(
+                spark, newTableIdentifier, externalDataFilesTableLocation)
+            .execute();
+    checkSnapshotIntegrity(
+        externalDataFilesTableLocation, externalDataFilesIdentifier, newTableIdentifier, result);
+    checkIcebergTableLocation(newTableIdentifier, externalDataFilesTableLocation);
+    checkDataFilePathsIntegrity(newTableIdentifier, externalDataFilesTableLocation);
+  }
+
+  @Test
+  public void testSnapshotSupportedTypes() {
+    String newTableIdentifier = destName(icebergCatalogName, snapshotTypeTestTableName);
+    SnapshotDeltaLakeTable.Result result =
+        DeltaLakeToIcebergMigrationSparkIntegration.snapshotDeltaLakeTable(
+                spark, newTableIdentifier, typeTestTableLocation)
+            .tableProperty(TableProperties.PARQUET_VECTORIZATION_ENABLED, "false")
```

Comment:
The `testTypeDataFrame` includes a timestamp column which is default to INT96 by delta lake/spark. 
According to #4196 and #4200 , currently the vectorized reader does not support `INT96`, so we must disable it.

I will also add this to the doc